web-common/src/main/java/org/openmrs/eip/web/contoller/BaseRestController.java

```diff
@@ -52,6 +53,41 @@ public Object doGet(Long id) {
 		    "jpa:" + getName() + "?query=SELECT c FROM " + getName() + " c WHERE c.id = " + id, null, getClazz());
 	}
 	
+	public Map<String, Object> doSearchByPeriod(String startDate, String endDate) {
+
+	    Map<String, Object> results = new HashMap<String, Object>(2);
+	    
+           String whereClause = StringUtils.EMPTY;
+            
+           if (!startDate.isBlank()) {
+                whereClause = " where c.dateCreated >= '" + startDate
+                        + "'";
+           }
+            
+           if (!endDate.isBlank()) {
+                whereClause += (whereClause.isBlank() ? " where " : " and ") + " c.dateCreated <= '" + endDate
+                        + "'";
+           }
+
+           Long count = on(camelContext).to("jpa:" + getName() + "?query=SELECT count(c) FROM " + getName()
+                   + " c " + whereClause).request(Long.class);
+            
+           if (count <= 0) {
```

Comment:
Actually I can't imagine count ever being less than 0,` if (count == 0)` should be the check

Reply:
Done